hm_pyhelper/tests/test_miner_param.py

```diff
@@ -0,0 +1,138 @@
+    },
+    {
+      "output": "timeout/retry error",
+      "result": "fail",
+      "test": "zone_locked(config)"
+    },
+    {
+      "output": "timeout/retry error",
+      "result": "fail",
+      "test": "slot_config(0..=15, ecc)"
+    },
+    {
+      "output": "timeout/retry error",
+      "result": "fail",
+      "test": "key_config(0..=15, ecc)"
+    },
+    {
+      "output": "timeout/retry error",
+      "result": "fail",
+      "test": "miner_key(0)"
+    }
+  ]
+
+class GatewayMfrProvisionMock:
+    def stdout():
+        return "example"
+
+class TestMinerParam(unittest.TestCase):
+
+    @patch('hm_pyhelper.miner_param.get_gateway_mfr_test_result', return_value = {
+            "result": "pass",
+            "tests": ALL_PASS_GATEWAY_MFR_TESTS
+        })
+    def test_provision_key_all_passed(self, get_gateway_mfr_test_result):
+        self.assertTrue(provision_key())
+        
+
+    @patch('subprocess.run', return_value = GatewayMfrProvisionMock())
+    @patch('hm_pyhelper.miner_param.get_gateway_mfr_test_result', return_value = {
+            "result": "fail",
+            "tests": NONE_PASS_GATEWAY_MFR_TESTS
+        })
+    def test_provision_key_none_passed(self, get_gateway_mfr_test_result, subprocess_run):
+        self.assertTrue(provision_key())
+
+
+    @patch('hm_pyhelper.miner_param.get_gateway_mfr_test_result', return_value = {
+            "result": "fail",
+            "tests": NONE_PASS_GATEWAY_MFR_TESTS
+        })
+    def test_did_gateway_mfr_test_result_include_miner_key_fail(self, get_gateway_mfr_test_result):
+        self.assertFalse(did_gateway_mfr_test_result_include_miner_key_pass(get_gateway_mfr_test_result))
+
+
+    def test_did_gateway_mfr_test_result_include_miner_key_pass(self):
+        get_gateway_mfr_test_result = {
+            "result": "fail",
+            "tests": [
+                {
+                "output": "timeout/retry error",
+                "result": "fail",
+                "test": "serial"
+                },
+                {
+                "output": "timeout/retry error",
+                "result": "fail",
+                "test": "zone_locked(data)"
+                },
+                {
+                "output": "timeout/retry error",
+                "result": "fail",
+                "test": "zone_locked(config)"
+                },
+                {
+                "output": "timeout/retry error",
+                "result": "fail",
+                "test": "slot_config(0..=15, ecc)"
+                },
+                {
+                "output": "timeout/retry error",
+                "result": "fail",
+                "test": "key_config(0..=15, ecc)"
+                },
+                {
+                "output": "ok",
+                "result": "pass",
+                "test": "miner_key(0)"
+                }
+            ]
+        }
+        self.assertTrue(did_gateway_mfr_test_result_include_miner_key_pass(get_gateway_mfr_test_result))
```

Comment:
Flagging this as potentially problematic. In this context, I think it is correct -- don't attempt to regen key if its already present, even if over parts of the ECC are acting up.